blockchain_parser/blockchain.py

```diff
@@ -103,13 +103,90 @@ def __getBlockIndexes(self, index):
             self.indexPath = index
         return self.blockIndexes
 
+    def _index_confirmed(self, chain_indexes, num_confirmations=6):
+        """Check if the first block index in "chain_indexes" has at least
+        "num_confirmation" (6) blocks built on top of it.
+        If it doesn't it is not confirmed and is an orphan.
+        """
+
+        # chains holds a 2D list of sequential block hash chains
+        # as soon as there an element of length num_confirmations,
+        # we can make a decision about whether or not the block in question
+        # is confirmed by checking if it's hash is in that list
+        chains = []
+        # this is the block in question
+        first_block = None
+
+        # loop through all future blocks
+        for i, index in enumerate(chain_indexes):
+            # if this block doesn't have data don't confirm the block in question
+            if index.file == -1 or index.data_pos == -1:
+                return False
+
+            # parse the block
+            blkFile = os.path.join(self.path, "blk%05d.dat" % index.file)
```

Comment:
Since the index object has all of the header data, we could optimize this part by hashing it instead of seeking the block. But given how many orphan blocks there is, it's overkill